src/utils/perfherder/chartJs/perfherderFormatter.js

```diff
@@ -7,33 +7,78 @@ const dataToChartJSformat = data => data.map(({ datetime, value }) => ({
   y: value,
 }));
 
-const generateInitialOptions = (meta) => {
-  const higherIsBetter = (meta.lower_is_better === false);
+const fetchPushidData = async (url) => {
+  try {
+    const response = await fetch(url);
+    const data = await response.json();
+    return data;
+  } catch (err) {
+    return err;
+  }
+};
+
+const generateInitialOptions = (series) => {
+  const higherIsBetter = !series.meta.lower_is_better;
+  const higherOrLower = higherIsBetter ? 'higher is better' : 'lower is better';
   return {
     reverse: higherIsBetter,
     scaleLabel: higherIsBetter ? 'Score' : 'Load time',
     tooltips: {
       callbacks: {
         footer: (tooltipItems, data) => {
+          const tooltipData = [];
           let delta = 'n/a';
+          let deltaPercentage = 'n/a';
+          let jobLink = '';
+          let hgLink = '';
+          let dataset = 'n/a';
+          let currentData = 'n/a';
           if (tooltipItems[0].index > 0) {
-            delta = (data.datasets[tooltipItems[0].datasetIndex].data[tooltipItems[0].index].y
-              - data.datasets[tooltipItems[0].datasetIndex].data[tooltipItems[0].index - 1].y)
-              .toFixed(2);
-            }
-            return `Delta: ${delta}`;
-          },
+            dataset = data.datasets[tooltipItems[0].datasetIndex].data;
+
+            currentData = dataset[tooltipItems[0].index].y;
+            const previousData = dataset[tooltipItems[0].index - 1].y;
+
+            delta = (currentData - previousData).toFixed(2);
+            deltaPercentage = (((currentData - previousData) / previousData) * 100).toFixed(2);
+
+            const tooltipDataSet = series.data[tooltipItems[0].index];
+            // make job link against job id
+            jobLink = `https://treeherder.mozilla.org/#/jobs?repo=mozilla-central&revision=ca0f00593e38cdab54c3a990059bbf1150e77365&selectedJob=${tooltipDataSet.job_id}&group_state=expanded`;
+
+            // generate hg link
+            const pushId = tooltipDataSet.push_id;
+            const previousPushID = series.data[tooltipItems[0].index - 1].push_id;
+
+            const pushiIdLink = `https://treeherder.mozilla.org/api/project/mozilla-central/resultset/${pushId}/?format=json`;
+            const previousPushiIdLink = `https://treeherder.mozilla.org/api/project/mozilla-central/resultset/${previousPushID}/?format=json`;
+
+            // get push data against pushid
+            const pushIdData = fetchPushidData(pushiIdLink);
+            const previousPushIdData = fetchPushidData(previousPushiIdLink);
+
+            const recentRevision = pushIdData.revision;
+            const previousRevision = previousPushIdData.revision;
```

Comment:
When I console.log({recentRevision, previousRevision}) it gives me required results which means that we have data in variables but over tooltip, it looks like that promise isn't resolved. It is confusing...

Reply:
I think the easiest solution would be to modify this `/performance/data` endpoint to return the revision hash so that you don't need to perform these fetches to begin with. Let me work on that (the endpoint belongs to the Treeherder codebase) and I'll let you know when the change has landed. It will likely be a few days since it needs to be reviewed and merged.